src/matrix.go

```diff
@@ -0,0 +1,153 @@
+/*
+Copyright 2022 The Matrix.org Foundation C.I.C.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"strings"
+
+	"maunium.net/go/mautrix"
+	"maunium.net/go/mautrix/event"
+)
+
+const localSessionID = "sfu"
+
+func InitMatrix() error {
+	client, err := mautrix.NewClient(config.HomeserverURL, config.UserID, config.AccessToken)
+	if err != nil {
+		log.Fatal("Failed to create client", err)
+	}
+
+	whoami, err := client.Whoami()
+	if err != nil {
+		log.Fatal("Failed to identify SFU user", err)
+	}
+	if config.UserID != whoami.UserID {
+		log.Fatalf("Access token is for the wrong user: %s", config.UserID)
+	}
+	log.Printf("Identified SFU as device %s", whoami.DeviceID)
+	client.DeviceID = whoami.DeviceID
+
+	focus := new(Focus)
+	focus.Init(fmt.Sprintf("%s (%s)", config.UserID, client.DeviceID))
+
+	syncer := client.Syncer.(*mautrix.DefaultSyncer)
+	syncer.ParseEventContent = true
+
+	// TODO: E2EE
+
+	getExistingCall := func(confID string, callID string) (*Call, error) {
+		var conf *Conference
+		var call *Call
+
+		if conf, err = focus.GetConf(confID, false); err != nil || conf == nil {
+			log.Printf("failed to get conf %s: %s", confID, err)
+			return nil, err
+		}
+		if call, err = conf.GetCall(callID, false); err != nil || call == nil {
+			log.Printf("failed to get call %s: %s", callID, err)
+			return nil, err
+		}
+		return call, nil
+	}
+
+	syncer.OnSync(func(resp *mautrix.RespSync, since string) bool {
+		for _, evt := range resp.ToDevice.Events {
```

Comment:
Is this really the only way to get todevice events in mautrix-go?

Reply:
Not sure, this was originally written by M - /me goes ask it the Maunium room